Set uvicorn server log level

peterroelants opened the pull request:
## Summary
Set graphql uvicorn server log level based on Prefect config.

Solves https://github.com/PrefectHQ/server/issues/216


## Importance
Allows the user to control the access logs of the Prefect GraphQL service (which can become quite large on the default `DEBUG` level).


## Checklist

This PR:

- [x] adds new tests (N/A)
- [x] adds a change file in the `changes/` directory (if appropriate)

madkinsz commented on src/prefect_server/services/graphql/server.py:
```diff
@@ -51,9 +51,24 @@ def health(request: Request) -> JSONResponse:
+    # Uvicorn log levels are lower case
+    uvicorn_log_level = prefect_log_level.lower()
+    if uvicorn_log_level in uvicorn.config.LOG_LEVELS:
+        logger.info(f"Using uvicorn log level = {repr(uvicorn_log_level)}")
```
You can do quick reprs in Python string formatting
```suggestion
        logger.info(f"Using uvicorn log level = {uvicorn_log_level!r}")
```

madkinsz commented on src/prefect_server/services/graphql/server.py:
```diff
@@ -51,9 +51,24 @@ def health(request: Request) -> JSONResponse:
+        return uvicorn_log_level
+    else:
+        logger.warning(
+            f"{repr(uvicorn_log_level)} not a valid uvicorn log level, falling back on default."
```
```suggestion
            f"{uvicorn_log_level!r} not a valid uvicorn log level, falling back on default."
```

madkinsz reviewed (commented):
This seems like a really reasonable fix to me, just some minor nits.

I think it is worth noting that:
- We can disable access logs to Uvicorn without changing the log level, this might be good to add to the config as `server.logging.log_access = false` and pass to Uvicorn
- We probably will increase the healthcheck interval because it's pretty fast

Thanks so much for contributing!

peterroelants commented:
@madkinsz Thank you for the quick review and the `repr` f-string suggestion, I didn't know about that.

I added a `server.logging.log_access` logging level as you suggested. However, I'm unsure if it should be at `logging.log_access`, or if it would be better to provide a `graphql.logging.log_access` config field. Do you have any opinion on this?

madkinsz commented:
> @madkinsz Thank you for the quick review and the `repr` f-string suggestion, I didn't know about that.

You're welcome :) Also, I left this as is before because I didn't think it was a big deal, but when using the config it's nice to use the attribute (`.`) access pattern e.g. `prefect_server.configuration.config.server.logging.level` instead of item access. They're equivalent but staying consistent with the rest of the repo makes it easier to search for usage of config values in the code.

> I added a `server.logging.log_access` logging level as you suggested. However, I'm unsure if it should be at `logging.log_access`, or if it would be better to provide a `graphql.logging.log_access` config field. Do you have any opinion on this?

Great point! This would fit better under `[services.graphql]`. Perhaps we should name it `services.graphql.disable_access_logs = True` to match the Uvicorn name.

> In the case of `graphql.logging.log_access` we might also want to change the logging level change to influence `graphql.logging.level`?

I think the rename suggested above will make it clear that you're just disabling it regardless of log level (as in Uvicorn). I don't think we should mutate another config value based on it.

peterroelants commented:
> when using the config it's nice to use the attribute (`.`) access pattern e.g. `prefect_server.configuration.config.server.logging.level` instead of item access. They're equivalent but staying consistent with the rest of the repo makes it easier to search for usage of config values in the code.

I see now you are using box. I updated the access pattern as you suggested.

> Great point! This would fit better under `[services.graphql]`. Perhaps we should name it `services.graphql.disable_access_logs = True` to match the Uvicorn name.

Good suggestion, I updated the code with this.

madkinsz reviewed (approved):
Thanks again!
